recipes/http-request/all/conanfile.py

```diff
@@ -0,0 +1,87 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd
+from conan.tools.files import copy, get
+from conan.tools.layout import basic_layout
+from conan.tools.scm import Version
+import os
+
+required_conan_version = ">=1.50.0"
+
+
+class CpphttplibConan(ConanFile):
+    name = "http-request"
+    description = "A modern C++ lightweight cross-platform HTTP request library."
+    license = "MIT"
+    homepage = "https://github.com/Nevermore1994/http-request"
+    url = "https://github.com/conan-io/conan-center-index"
+    topics = ("http", "https", "modern C++")
+    package_type = "header-library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "with_openssl": [True, False]
+    }
+    default_options = {
+        "with_openssl": True,
+    }
+    no_copy_source = True
+
+    def requirements(self):
+        if self.options.with_openssl:
+            self.requires("openssl/[>=1.1 <4]")
```

Comment:
https://github.com/Nevermore1994/http-request#notes seems to note that openssl is required in Windows and not optional. Is this right? Should this not be an option to disable it on Windows? Or how does this work? Thanks!